src/components/UserProfile/UserProfileSettings/index.js~HEAD

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { withStyles } from "@material-ui/styles";
+import Grid from "@material-ui/core/Grid";
+import TextField from "@material-ui/core/TextField";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import { connect } from "react-redux";
+
+import ErrorMsgBox from "../../common/ErrorMsgBox";
+import StyledButton from "../../common/StyledButton";
+import { useStyles } from "./styles";
+import Session from "../../../utility/constants/Session";
+import { userActions } from "../../../Redux/actionCreators";
+import Routes from "../../../utility/constants/Routes";
+
+class UserProfileSettings extends Component {
+  state = {
+    error: undefined,
+  };
+
+  handleEmailChange = event => {
+    this.setState({
+      email: event.target.value,
+    });
+  };
+
+  handleEmailAlerts = () => {
+    this.setState(prevState => ({ emailAlerts: !prevState.emailAlerts }));
+  };
+
+  handleDelete = async () => {
+    try {
+      await this.props.deleteUserAccount();
+      alert("user profile delted successfully");
+      this.props.history.push(`/${Routes.AI_MARKETPLACE}`);
+    } catch (err) {
+      this.setState({ error: String(err) });
+    }
+  };
+
+  render() {
+    const { classes, email } = this.props;
+    const { error } = this.state;
+    const userName = sessionStorage.getItem(Session.USERNAME);
```

Comment:
We could start using redux instead of sessionStorage.

Reply:
done.
[commit](https://github.com/singnet/refactored-dapp/pull/17/commits/153ff3d5ff30301a1aecfbc017223b9445226ebd)
[commit](https://github.com/singnet/refactored-dapp/pull/17/commits/96c5026a3d7f473e129d1fb954c80b7bc9a950fd)